### PR TITLE
Generic FQDN for 4G/5G

### PR DESCRIPTION
**Pull request by orion-belt**

### CODE REVIEW

File: etc/spgw_u.conf [hunk 1]
```diff
@@ -20,7 +20,8 @@
 ################################################################################
 SPGW-U =
 {
-    FQDN = "gw@GW_ID@.spgw.node.epc.mnc@MNC03@.mcc@MCC@.@REALM@";
+    #FQDN = "gw@GW_ID@.spgw.node.epc.mnc@MNC03@.mcc@MCC@.@REALM@";
```
Comment (rdefosse) [inline etc/spgw_u.conf:23]: You can remove the commented line.

Or better you can explain both formats as comment (the 4G case and the 5G case)

**Review by rdefosse — CHANGES_REQUESTED** [review]

**Comment by thinhnt1983** [comment]
Hi @orion-belt it would be better if you can still keep FQDN = "gw@GW_ID@.spgw.node.epc.mnc@MNC03@.mcc@MCC@.@REALM@"; at the beginning of the conf file, since it's for 4G. and we shouldn't touch 4G part. Then you can add a new variable, e.g., FQDN_UPF inside 5G session. Then we process according to 5g flag. What do you think?

**Comment by orion-belt** [comment]
@thinhnt1983 @rdefosse 
Recommendations are fixed in latest commit.

File: etc/spgw_u.conf [hunk 2]
```diff
@@ -118,7 +118,8 @@ SPGW-U =
        ENABLE_5G_FEATURES = "@ENABLE_5G_FEATURES@" # Set to 'yes' to support 5G Features
        REGISTER_NRF = "@REGISTER_NRF@";  # Set to 'yes' if UPF resgisters to an NRF
        USE_FQDN_NRF = "@USE_FQDN_NRF@"; # Set to 'yes' if UPF relies on a DNS/FQDN service to resolve NRF's FQDN
- 
+       UPF_FQDN_5G  = "@UPF_FQDN_5G@";  #Set FQDN of UPF
```
Comment (rdefosse) [inline etc/spgw_u.conf:121]: @orion-belt @thinhnt1983 

This wlll require to modify at least a few docker-compose in the federation repository.

- `https://github.com/OPENAIRINTERFACE/openair-epc-fed/blob/master/docker-compose/magma-mme-demo/docker-compose.yml`
- `https://github.com/OPENAIRINTERFACE/openair-epc-fed/blob/master/docker-compose/oai-mme-legacy/docker-compose.yml`
- `https://github.com/OPENAIRINTERFACE/openair-epc-fed/blob/master/ci-scripts/dsTesterDockerCompose/docker-compose.tplt`
- `https://github.com/OPENAIRINTERFACE/openair-epc-fed/blob/master/ci-scripts/dsTesterDockerComposeLegacy/docker-compose.tplt`
- `https://github.com/OPENAIRINTERFACE/openair-epc-fed/blob/master/ci-scripts/oai-ran-sanity-check/spgwu.env`

Unless you put a default value in the `entrypoint.sh` like I did for the previous 5G fields.

Raphael

**Review by rdefosse — APPROVED** [review]

**Review by thinhnt1983 — APPROVED** [review]

**Pull request merged**